internal/app/go-filecoin/connectors/retrieval_market/client.go

```diff
@@ -49,28 +50,35 @@ func NewRetrievalClientConnector(
 }
 
 // GetOrCreatePaymentChannel gets or creates a payment channel and posts to chain
-func (r *RetrievalClientConnector) GetOrCreatePaymentChannel(ctx context.Context, clientAddress address.Address, minerAddress address.Address, clientFundsAvailable abi.TokenAmount, tok shared.TipSetToken) (address.Address, error) {
+func (r *RetrievalClientConnector) GetOrCreatePaymentChannel(ctx context.Context, clientAddress address.Address, minerAddress address.Address, clientFundsAvailable abi.TokenAmount, tok shared.TipSetToken) (address.Address, cid.Cid, error) {
+
+	errReturn := func(err error) (address.Address, cid.Cid, error) {
```

Comment:
Yeah, I was on the fence about it, but I thought I'd throw it out there. 
Making it a free function isn't great because you'd need one for every single set of return values.